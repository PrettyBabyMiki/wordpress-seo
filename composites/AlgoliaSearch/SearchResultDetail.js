import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { defineMessages, injectIntl, intlShape } from "react-intl";

import ArticleContent from "./ArticleContent";
import { YoastButton } from "../Plugin/Shared/components/YoastButton";
import { YoastLinkButton } from "../Plugin/Shared/components/YoastLinkButton";
import SvgIcon from "../Plugin/Shared/components/SvgIcon";
import { makeOutboundLink } from "../../utils/makeOutboundLink";
import breakpoints from "../../style-guide/responsive-breakpoints.json";
import colors from "../../style-guide/colors.json";

const messages = defineMessages( {
	searchResult: {
		id: "searchResultDetail.searchResult",
		defaultMessage: "Search result",
	},
	openButton: {
		id: "searchResultDetail.openButton",
		defaultMessage: "View in KB",
	},
	openButtonLabel: {
		id: "searchResultDetail.openButtonLabel",
		defaultMessage: "Open the knowledge base article in a new window or read it in the iframe below",
	},
	backButton: {
		id: "searchResultDetail.backButton",
		defaultMessage: "Go back",
	},
	backButtonLabel: {
		id: "searchResultDetail.backButtonLabel",
		defaultMessage: "Go back to the search results",
	},
	iframeTitle: {
		id: "searchResultDetail.iframeTitle",
		defaultMessage: "Knowledge base article",
	},
} );

const Detail = styled.section`
	outline: none;

	@media screen and ( max-width: ${ breakpoints.mobile } ) {
		margin: 0 -16px;
	}
`;

const Nav = styled.nav`
	padding: 0 16px 16px;
`;

const RightYoastLinkButton = makeOutboundLink( styled( YoastLinkButton )`
	float: right;
` );

/**
 * Create the JSX to render the SearchResultDetail component.
 *
 * @param {object} props The React props.
 *
 * @returns {ReactElement} A SearchResultDetail component.
 */
class SearchResultDetail extends React.Component {
	/**
	 * Creates navigational elements.
	 *
	 * @returns {ReactElement} The rendered element.
	 */
	createNavigation() {
		const formatMessage = this.props.intl.formatMessage;
		const openButtonText = formatMessage( messages.openButton );
		const openButtonLabel = formatMessage( messages.openButtonLabel );
		const backButtonText = formatMessage( messages.backButton );
		const backButtonLabel = formatMessage( messages.backButtonLabel );
		return (
			<Nav>
				<YoastButton aria-label={ backButtonLabel } onClick={ this.props.onBackButtonClicked }>
					<SvgIcon
						size="24px"
						color={ colors.$color_white }
						icon="angle-left"
					/>
					{ backButtonText }
				</YoastButton>
				<RightYoastLinkButton
					href={ this.props.post.permalink }
					aria-label={ openButtonLabel }
					target="_blank"
					rel="noopener noreferrer"
				>
					{ openButtonText }
					<SvgIcon
						size="24px"
						color={ colors.$color_white }
						icon="angle-right"
					/>
				</RightYoastLinkButton>
			</Nav>
		);
	}

	/**
	 * Renders the search result detail.
	 *
	 * @returns {ReactElement} The rendered element.
	 */
	render() {
		const formatMessage = this.props.intl.formatMessage;
		const searchResulLabel = formatMessage( messages.searchResult );
		const iframeTitle = formatMessage( messages.iframeTitle );
		return (
			<Detail
				aria-label={ searchResulLabel }
				tabIndex="-1"
				innerRef={ ( el ) => {
					this.detailWrapper = el;
				} }
			>
				{ this.createNavigation() }
				<ArticleContent post={ this.props.post } title={ iframeTitle } />
			</Detail>
		);
	}

	/**
	 * When the component mounts, set focus on the search detail wrapper.
	 *
	 * @returns {void}
	 */
	componentDidMount() {
		if ( this.detailWrapper !== null ) {
			this.detailWrapper.focus();
		}
	}
}

SearchResultDetail.propTypes = {
	post: PropTypes.object.isRequired,
	onBackButtonClicked: PropTypes.func.isRequired,
	intl: intlShape.isRequired,
};

export default injectIntl( SearchResultDetail );
