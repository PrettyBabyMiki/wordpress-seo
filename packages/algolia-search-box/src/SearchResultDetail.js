/* External imports */
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { defineMessages, injectIntl, intlShape } from "react-intl";

/* Yoast dependencies */
import { YoastButton, SvgIcon, YoastLinkButton } from "@yoast/components";
import { makeOutboundLink } from "@yoast/helpers";
import { colors, breakpoints } from "@yoast/style-guide";

/* Internal imports */
import ArticleContent from "./ArticleContent";

const messages = defineMessages( {
	searchResult: {
		id: "searchResultDetail.searchResult",
		defaultMessage: "Search result",
	},
	openButton: {
		id: "searchResultDetail.openButton",
		defaultMessage: "View in KB",
	},
	openButtonLabel: {
		id: "searchResultDetail.openButtonLabel",
		defaultMessage: "Open the knowledge base article in a new window or read it in the iframe below",
	},
	backButton: {
		id: "searchResultDetail.backButton",
		defaultMessage: "Go back",
	},
	backButtonLabel: {
		id: "searchResultDetail.backButtonLabel",
		defaultMessage: "Go back to the search results",
	},
	iframeTitle: {
		id: "searchResultDetail.iframeTitle",
		defaultMessage: "Knowledge base article",
	},
} );

const Detail = styled.section`
	outline: none;

	@media screen and ( max-width: ${ breakpoints.mobile } ) {
		margin: 0 -16px;
	}
`;

const Nav = styled.nav`
	padding: 0 16px 16px;
`;

const RightYoastLinkButton = makeOutboundLink( styled( YoastLinkButton )`
	float: right;
` );

/**
 * Create the JSX to render the SearchResultDetail component.
 *
 * @param {object} props The React props.
 *
 * @returns {ReactElement} A SearchResultDetail component.
 */
class SearchResultDetail extends React.Component {
	/**
	 * Creates navigational elements.
	 *
	 * @returns {ReactElement} The rendered element.
	 */
	createNavigation() {
		const formatMessage = this.props.intl.formatMessage;
		const openButtonText = formatMessage( messages.openButton );
		const openButtonLabel = formatMessage( messages.openButtonLabel );
		const backButtonText = formatMessage( messages.backButton );
		const backButtonLabel = formatMessage( messages.backButtonLabel );
		return (
			<Nav>
				<YoastButton aria-label={ backButtonLabel } onClick={ this.props.onBackButtonClicked }>
					<SvgIcon
						size="24px"
						color={ colors.$color_white }
						icon="angle-left"
					/>
					{ backButtonText }
				</YoastButton>
				<RightYoastLinkButton
					href={ this.props.post.permalink }
					aria-label={ openButtonLabel }
				>
					{ openButtonText }
					<SvgIcon
						size="24px"
						color={ colors.$color_white }
						icon="angle-right"
					/>
				</RightYoastLinkButton>
			</Nav>
		);
	}

	/**
	 * Renders the search result detail.
	 *
	 * @returns {ReactElement} The rendered element.
	 */
	render() {
		const formatMessage = this.props.intl.formatMessage;
		const searchResulLabel = formatMessage( messages.searchResult );
		const iframeTitle = formatMessage( messages.iframeTitle );
		return (
			<Detail
				aria-label={ searchResulLabel }
				tabIndex="-1"
				ref={ ( el ) => {
					this.detailWrapper = el;
				} }
			>
				{ this.createNavigation() }
				<ArticleContent post={ this.props.post } title={ iframeTitle } />
			</Detail>
		);
	}

	/**
	 * When the component mounts, set focus on the search detail wrapper.
	 *
	 * @returns {void}
	 */
	componentDidMount() {
		if ( this.detailWrapper !== null ) {
			this.detailWrapper.focus();
		}
	}
}

SearchResultDetail.propTypes = {
	post: PropTypes.object.isRequired,
	onBackButtonClicked: PropTypes.func.isRequired,
	intl: intlShape.isRequired,
};

export default injectIntl( SearchResultDetail );
