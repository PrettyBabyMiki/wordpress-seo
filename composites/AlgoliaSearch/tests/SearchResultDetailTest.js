import React from "react";
import renderer from "react-test-renderer";

import SearchResultDetail from "../SearchResultDetail.js";
let post = { permalink: "https://kb.yoast.com/kb/passive-voice/", post_title: "Post Title", objectID: 1 };

test( "the SearchResultDetail component matches the snapshot", () => {
	const component = renderer.create(
		<SearchResultDetail post={ post } showDetail={ () => {} } onClick={ () => {} } iframeTitle="Title"/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
