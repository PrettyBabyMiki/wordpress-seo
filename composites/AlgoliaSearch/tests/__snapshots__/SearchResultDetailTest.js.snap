// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the SearchResultDetail component matches the snapshot 1`] = `
<section
  className="sc-bZQynM bOrPFB"
>
  <nav
    className="sc-gzVnrw kJqUwD"
  >
    <button
      aria-expanded={undefined}
      className="sc-htpNat bmCLLV sc-bwzfXH dIZeqH"
      onClick={[Function]}
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          className="sc-dnqmqq UPict"
          focusable="false"
          role="img"
        />
        Go back
      </span>
    </button>
    <a
      aria-label="Open the knowledge base article in a new window or read it in the iframe below"
      className="sc-htoDjs hpJeHy sc-ifAKCX ifSeLk sc-bxivhb ccVExY"
      href="https://kb.yoast.com/kb/passive-voice/"
      rel="noopener noreferrer"
      target="_blank"
    >
      View in KB
      <svg
        aria-hidden="true"
        className="sc-iwsKbI cCpRGs"
        focusable="false"
        role="img"
      />
      <span
        className="sc-EHOje dtuOhP"
      >
        (Opens in a new browser tab)
      </span>
    </a>
  </nav>
  <iframe
    className="kb-search-content-frame sc-bdVaJa jsOWvL"
    src="https://kb.yoast.com/kb/passive-voice/amp/?source=wpseo-kb-search"
    title="Knowledge base article"
  />
</section>
`;
